### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by sncalvo**
### **What/How I did it**
I added a comment model and comments controller with partials to use them in articles views.

Comments can now be created, read and destroyed.

### **How to verify it**
Go to articles show action and try links and buttons for respective operations.

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
+    @comment = @article.comments.find(params[:id])
+    @comment.destroy
+    redirect_to article_path(@article)
+  end
+
+  private
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: Add extra line

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
+    @article = Article.find(params[:article_id])
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: You don't need instance variables here

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    @comment = @article.comments.create(comment_params)
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    @article = Article.find(params[:article_id])
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: Same, you don't need instance variables here

File: app/helpers/articles_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module ArticlesHelper
```
Comment (ignaciogrondona) [inline app/helpers/articles_helper.rb]: Remove unnecessary helper

File: app/views/articles/show.html.erb [hunk 5]
```diff
@@ -1,9 +1,17 @@
 <p>
   <h4>Title: </h4>
   <%= @article.title %>
+</p>
```
Comment (ignaciogrondona) [inline app/views/articles/show.html.erb]: I don't think this should go inside p tag.

The p tag defines a paragraph:

https://www.w3schools.com/tags/tag_p.asp

File: app/views/comments/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<%= form_with(model: [ @article, @article.comments.build ], local: true) do |form| %>
```
Comment (ignaciogrondona) [inline app/views/comments/_form.html.erb]: remove extra spaces

File: app/views/comments/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  <p>
```
Comment (ignaciogrondona) [inline app/views/comments/_form.html.erb]: Same for p tags

File: test/fixtures/comments.yml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (ignaciogrondona) [inline test/fixtures/comments.yml]: Remove unused files

File: test/controllers/articles_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (ignaciogrondona) [inline test/controllers/articles_controller_test.rb]: remove

File: test/models/comment_test.rb [hunk 10]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
```
Comment (ignaciogrondona) [inline test/models/comment_test.rb]: remove

File: app/controllers/comments_controller.rb [hunk 11]
```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+  def create
+    article = Article.find(params[:article_id])
+    @comment = article.comments.create(comment_params)
+    redirect_to article_path(@article)
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: yo need to use the article variable now

File: app/controllers/comments_controller.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
+class CommentsController < ApplicationController
+  def create
+    article = Article.find(params[:article_id])
+    @comment = article.comments.create(comment_params)
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: no need for instance variable here either

File: app/controllers/comments_controller.rb [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    article = Article.find(params[:article_id])
+    @comment = article.comments.find(params[:id])
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: no need for instance variable

File: app/controllers/comments_controller.rb (hunk 12 above)
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: What if the create fails? We need to check for errors

File: app/controllers/comments_controller.rb [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+    @comment.destroy
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: Check errors here too

File: app/views/comments/_comment.html.erb [hunk 15]
```diff
@@ -0,0 +1,15 @@
+<p>
+  <strong>Commenter:</strong>
+  <%= comment.commenter %>
+</p>
+
+<p>
```
Comment (ignaciogrondona) [inline app/views/comments/_comment.html.erb]: Please avoid using p tag here too.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (sncalvo) [reply app/controllers/comments_controller.rb]: After private?

File: app/controllers/articles_controller.rb [hunk 16]
```diff
@@ -39,8 +39,11 @@ def update
   def destroy
     @article = Article.find(params[:id])
     @article.destroy
-
-    redirect_to articles_path
+    if @article.destroyed?
```
Comment (ignaciogrondona) [inline app/controllers/articles_controller.rb]: `if @article.destroy `

File: app/controllers/comments_controller.rb [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+    else
+      flash[:error] = 'Comment could not be saved'
+      redirect_to article_path(article)
+    end
+  end
+
+  def destroy
+    article = Article.find(params[:article_id])
+    comment = article.comments.find(params[:id])
+    comment.destroy
```
Comment (ignaciogrondona) [inline app/controllers/comments_controller.rb]: same here

**Pull request merged**